macOS build errors

### What is the issue?

As @nevack [noted](https://github.com/transmission/transmission/pull/4147#issuecomment-1312572394), confirmed here, macOS Transmission cannot be built after #4147.

`		</array>
	</dict>
	<key>com.apple.actool.notices</key>
	<array>
		<dict>
			<key>description</key>
			<string>Failed to read file attributes for "/Users/gary/Documents/GitHub/Transmission-Primary/macosx/AppIcon"</string>
			<key>failure-reason</key>
			<string>No such file or directory</string>
		</dict>
	</array>
</dict>
</plist>
Copying OS X content macosx/Transmission.app/Contents/Resources/AppIcon.icns
Error copying file "/Users/gary/Documents/GitHub/Transmission-Primary/build/macosx/AppIcon.icns" to "macosx/Transmission.app/Contents/Resources/AppIcon.icns".
make[2]: *** [macosx/Transmission.app/Contents/Resources/AppIcon.icns] Error 1
make[1]: *** [macosx/CMakeFiles/transmission-mac.dir/all] Error 2
make: *** [all] Error 2`

### Which application of Transmission?

macOS app

### Which version of Transmission?

_No response_

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -199,7 +199,6 @@ jobs:
             -DCMAKE_BUILD_TYPE=RelWithDebInfo \
             -DCMAKE_INSTALL_PREFIX=pfx \
             -DCMAKE_OSX_ARCHITECTURES='x86_64' \
-            -DCMAKE_OSX_DEPLOYMENT_TARGET='10.10' \
             -DCMAKE_PREFIX_PATH=`brew --prefix`/opt/qt@5 \
             -DENABLE_CLI=${{ (needs.what-to-make.outputs.make-cli == 'true') && 'ON' || 'OFF' }} \
             -DENABLE_DAEMON=${{ (needs.what-to-make.outputs.make-daemon == 'true') && 'ON' || 'OFF' }} \
@@ -369,7 +368,6 @@ jobs:
             -DCMAKE_BUILD_TYPE=RelWithDebInfo \
             -DCMAKE_INSTALL_PREFIX=pfx \
             -DCMAKE_OSX_ARCHITECTURES='x86_64' \
-            -DCMAKE_OSX_DEPLOYMENT_TARGET='10.10' \
             -DCMAKE_PREFIX_PATH=`brew --prefix`/opt/qt@5 \
             -DENABLE_CLI=${{ (needs.what-to-make.outputs.make-cli == 'true') && 'ON' || 'OFF' }} \
             -DENABLE_DAEMON=${{ (needs.what-to-make.outputs.make-daemon == 'true') && 'ON' || 'OFF' }} \

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,8 +6,8 @@ endif()
 
 # The value of this variable should be set prior to the first project() command invocation.
 # See: https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html
-if(CMAKE_OSX_DEPLOYMENT_TARGET STREQUAL "")
-    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13")
+if(NOT CMAKE_OSX_DEPLOYMENT_TARGET)
+    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13" CACHE STRING "Minimum OS X version to target for deployment" FORCE)
 endif()
 
 project(transmission)

--- a/macosx/CMakeLists.txt
+++ b/macosx/CMakeLists.txt
@@ -1,5 +1,10 @@
 project(trmac)
 
+if(CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS 10.9)
+    # 10.9 required for actool to produce Assets.car
+    message(FATAL_ERROR "Targeting macOS versions before 10.9 is not supported, please adjust CMAKE_OSX_DEPLOYMENT_TARGET (${CMAKE_OSX_DEPLOYMENT_TARGET})")
+endif()
+
 include_directories(${CMAKE_SOURCE_DIR})
 
 add_compile_options(

